MoreLinq.Test/MoveTest.cs

```diff
@@ -70,9 +71,14 @@ public static IEnumerable<object> MoveSource()
         {
             const int length = 10;
 
-            return from index in Enumerable.Range(0, length)
-                   from count in Enumerable.Range(0, length + 1)
-                   select new TestCaseData(length, index, count, index);
+            foreach (var index in Enumerable.Range(0, length))
```

Comment:
You can keep this as a query (instead of turning it into an iterator):

```c#
return from index in Enumerable.Range(0, length)
       from count in Enumerable.Range(0, length + 1)
       from tcd in new[]
       {
           new TestCaseData(length, index, count, Math.Max(0, index - 1)),
           new TestCaseData(length, index, count, index + 1),
       }
       select tcd;
```

This is more powerful because (in a later refactoring, after this PR) you can refactor the test data into a field:

```c#
public static IEnumerable<TestCaseData> MoveSource =>
    from length in new[] { 10 }
    from index in Enumerable.Range(0, length)
    from count in Enumerable.Range(0, length + 1)
    from tcd in new[]
    {
        new TestCaseData(length, index, count, Math.Max(0, index - 1)),
        new TestCaseData(length, index, count, index + 1),
    }
    select tcd;
```